docs/architecture/adr/0026-dotnet-dependency-injection-enhancements.md

```diff
@@ -0,0 +1,154 @@
+---
+adr: "0026"
+status: "Proposed"
+date: 2025-05-30
+tags: [server]
+---
+
+# 0026 - Adopt `TryAdd` Dependency Injection Overloads
+
+<AdrTable frontMatter={frontMatter}></AdrTable>
+
+## Context and Problem Statement
+
+`Microsoft.Extensions.DependencyInjection` (the DI provider we use) has last one wins behavior. This
+means that if you inject two services of the same service type the last implementation that was
+registered wins. For example:
+
+```csharp
+services.AddSingleton<IMyService, ImplementationOne>();
+// Somewhere later on in the codebase
+services.AddSingleton<IMyService, ImplementationTwo>();
+```
+
+when a service or controller inject `IMyService` they will be getting `ImplementationTwo` even
+though `ImplementationOne` _does_ still exist in the service container. It exists in the container
+still because if you were to instead inject `IEnumerable<IMyService>` you would receive an
+enumerable containing 2 services, one for each implementation that was registered. This **is** the
+behavior you want sometimes but it is much rarer to inject an enumerable of services as opposed to
+injecting just a single service. This is where the [`TryAdd`][try-add-definitions] overloads on
+`IServiceCollection` (in the `Microsoft.Extensions.DependencyInjection.Extensions` namespace) come
+in handy. They allow you to more explicitly declare the expected usage of a service during service
```

Comment:
```suggestion
in handy. One can more explicitly declare the expected usage of a service during service
```
There's a lot of "this is" and "they" running each sentence structure so I am trying to break that up with this.